src/cli/study-backrefs.js

```diff
@@ -0,0 +1,483 @@
+  "resource-timing-1": "resource-timing",
+  "selectors-3": "selectors",
+  "selectors4": "selectors",
+  "ServiceWorker": "service-workers",
+  "wai-aria-1.1": "wai-aria-1.2",
+  "wasm-core-1": "wasm-core",
+  "webauthn-1": "webauthn",
+  "webdriver": "webdriver2",
+  "webdriver1": "webdriver2"
+};
+
+// TODO: check the link is non-normative (somehow)
+const shortnameOfNonNormativeDocs = [
+    "accept-encoding-range-test",
+  "aria-practices",
+  "Audio-EQ-Cookbook",
+  "books",
+  "capability-urls",
+  "clreq",
+  "css-2017",
+  "css-print",
+  "css3-marquee",
+  "css3-preslev",
+  "design-principles",
+  "discovery-api",
+  "dpub-latinreq",
+  "dpub-pagination",
+  "file-system-api",
+  "fingerprinting-guidance",
+  "html-design-principles",
+  "ilreq",
+  "installable-webapps",
+  "jlreq",
+  "klreq",
+  "media-accessibility-reqs",
+  "media-source-testcoverage",
+  "motion-sensors",
+  "predefined-counter-styles",
+  "rdf11-primer",
+  "security-privacy-questionnaire",
+  "security-questionnaire",
+  "sensor-polyfills",
+  "sensors",
+  "sniffly",
+  "spatial-navigation",
+  "ssml-sayas",
+  "storage-partitioning",
+  "streamproc",
+  "touch-events-extensions",
+  "typography",
+  "using-aria",
+  "wai-aria-implementation",
+  "wai-aria-practices",
+  "wai-aria-practices-1.1",
+  "wai-aria-practices-1.2",
+  "wai-aria-roadmap",
+  "wake-lock-use-cases",
+  "web-audio-perf",
+  "web-intents",
+  "webaudio-usecases",
+  "webdatabase",
+  "webrtc-interop-reports",
+  "webrtc-nv-use-cases"
+];
+
+
+function studyCrawlResults(edResults, trResults) {
+  const report = {};
+  const missingSpecs = {};
+
+  function recordAnomaly(spec, anomalyType, link) {
+    if (!report[spec.url]) {
+      report[spec.url] = {
+        title: spec.title,
+        notExported: [],
+        notDfn: [],
+        brokenLink: [],
+        evolvingLink: [],
+        outdatedSpec: [],
+        unknownSpec: [],
+        datedUrl: []
+      };
+    }
+    report[spec.url][anomalyType].push(link);
+  }
+
+  function recordUnknownSpec(link, shortname, spec) {
+    if (!missingSpecs[shortname]) {
+      missingSpecs[shortname] = [];
+    }
+    missingSpecs[shortname].push({link, spec: spec.url});
+  }
+
+  edResults.forEach(spec => {
+    Object.keys(spec.links)
+      .filter(matchSpecUrl)
+      .forEach(link => {
+        let shortname;
+        let nakedLink = link;
+        if (nakedLink.endsWith(".html")) {
+          nakedLink = nakedLink.replace(/\/[^/]*\.html$/, '/');
+        }
+        if (nakedLink[nakedLink.length - 1] !== '/') {
+          nakedLink += '/';
+        }
+        shortname = (edResults.find(r =>
+          r.url === nakedLink ||
+          (r.release && r.release.url === nakedLink) ||
+          r.nightly.url === nakedLink ||
+          (r.series && nakedLink === 'https://www.w3.org/TR/' + r.series.shortname + '/') ) || {}).shortname;
+        if (!shortname) {
+          try {
+            ({shortname} = computeShortname(link));
```

Comment:
Talking about code obfuscation, this line does a very good job at pretending that it does something but actually doing something else. The `computeShortname` function returns a string, so this always sets `shortname` to undefined, which is probably not what was intended.

The logic of the few next lines remained somewhat good, except that the changes I made to make things more logic actually broke it. Anyway, I'll fix that.